common/src/Microsoft.Azure.IIoT.Hub.Processor/src/Services/EventProcessorFactory.cs

```diff
@@ -308,7 +330,17 @@ private class EventProperties : IDictionary<string, string> {
             private static readonly Gauge kEventProcessorDetails = Metrics
                 .CreateGauge("iiot_event_processor_info", "details about event processor",
                     new GaugeConfiguration {
-                        LabelNames = new[] { "id", "partition_id", "status" }
+                        LabelNames = new[] { "id", "eventhub_name", "consumer_name", "partition_id", "status" }
+                    });
+            private static readonly Gauge kOldEvent = Metrics
+                .CreateGauge("iiot_event_processor_skipped_events", "number of messages skipped due to being old",
+                    new GaugeConfiguration {
+                        LabelNames = new[] { "id", "eventhub_name", "consumer_name", "partition_id" }
+                    });
+            private static readonly Gauge kEventProcessorMessages = Metrics
+                .CreateGauge("iiot_event_processor_message_count", "number of messages processed",
+                    new GaugeConfiguration {
+                        LabelNames = new[] { "id", "eventhub_name", "consumer_name", "partition_id", "read_timestamp_utc" }
```

Comment:
I still cannot understand why we need to add the timestamp as a label and how will that be visualized for showing number of processed messages. Well, mostly I do not understand what additional functionality 3 hour slicing going to provide compared to just having a gauge which counts number of processed events. What we need is a way to understand for a given dynamic interval how many messages have been processed through each event processor and then just visualize them in one line chart.

Reply:
To understand for a given dynamic interval how many messages have been processed through each event processor, we would require the actual timestamp for the metrics. For starter, we can rely on Prometheus/OMS agent publish timestamp being equal to actual event read timestamp.